### PR TITLE
nightly run new versions and collect pypi in python

### PR DESCRIPTION
**Pull request by nirLumigo**

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -20,32 +96,25 @@ def python_versions() -> Union[List[str], bool]:
...
-            for version in all_versions:
-                parsed_version = parse_version(version)
-                minor = f"{parsed_version.major}.{parsed_version.minor}"
-                if minor_to_version.get(minor, parse_version("0")) < parsed_version:
-                    minor_to_version[minor] = parsed_version
-            minor_versions = [v.public for v in minor_to_version.values()]
-            print("running minor versions of", dependency_name, minor_versions)
-            return minor_versions
-    except FileNotFoundError:
-        return []
```
Comment (nirLumigo) [inline noxfile.py:48]: I think it can cause a bug that test will pass although they didn't run so I deleted it.

File: noxfile.py [hunk 2]
```diff
@@ -56,63 +125,60 @@ def integration_tests_boto3(
...
-                for proc in psutil.process_iter():
-                    # The python process is names "Python" os OS X and "uvicorn" on CircleCI
-                    if proc.name() == "uvicorn":
-                        proc.kill()
-                    elif proc.name().lower() == "python":
-                        cmdline = proc.cmdline()
-                        if len(cmdline) > 1 and cmdline[1].endswith("/uvicorn"):
+    with TestedVersions.save_tests_result("boto3", "boto3", boto3_version):
+        install_package("boto3", boto3_version, session)
+
```
Comment (nirLumigo) [inline noxfile.py:130]: This is only indentation change

File: noxfile.py [hunk 3]
```diff
@@ -222,63 +281,60 @@ def integration_tests_fastapi(
...
-                for proc in psutil.process_iter():
-                    # The python process is names "Python" os OS X and "flask" on CircleCI
-                    if proc.name() == "flask":
-                        proc.kill()
-                    elif proc.name().lower() == "python":
-                        cmdline = proc.cmdline()
-                        if len(cmdline) > 1 and cmdline[1].endswith("/flask"):
+    with TestedVersions.save_tests_result("flask", "flask", flask_version):
+        install_package("flask", flask_version, session)
+
```
Comment (nirLumigo) [inline noxfile.py:286]: This is only an indentation change

File: noxfile.py [hunk 4]
```diff
@@ -290,63 +346,60 @@ def integration_tests_pymongo(
...
-                for proc in psutil.process_iter():
-                    # The python process is names "Python" os OS X and "uvicorn" on CircleCI
-                    if proc.name() == "uvicorn":
-                        proc.kill()
-                    elif proc.name().lower() == "python":
-                        cmdline = proc.cmdline()
-                        if len(cmdline) > 1 and cmdline[1].endswith("/uvicorn"):
+    with TestedVersions.save_tests_result("pymongo", "pymongo", pymongo_version):
+        install_package("pymongo", pymongo_version, session)
+
```
Comment (nirLumigo) [inline noxfile.py:351]: This is only an indentation change

File: noxfile.py [hunk 5]
```diff
@@ -358,60 +411,57 @@ def integration_tests_pymysql(
...
-                for proc in psutil.process_iter():
-                    # The python process is names "Python" os OS X and "uvicorn" on CircleCI
-                    if proc.name() == "uvicorn":
-                        proc.kill()
-                    elif proc.name().lower() == "python":
-                        cmdline = proc.cmdline()
-                        if len(cmdline) > 1 and cmdline[1].endswith("/uvicorn"):
+    with TestedVersions.save_tests_result("pymysql", "pymysql", pymysql_version):
+        install_package("PyMySQL", pymysql_version, session)
+
```
Comment (nirLumigo) [inline noxfile.py:416]: This is only an indentation change

**Review by GuyMoses — APPROVED** [review]

**Pull request merged**